### ISSUE TITLE
No module named imp

### ISSUE THREAD

**Issue by liquidsec** [richardpenman/whois#204]
Sorry for the slightly obnoxious approach of reopening a closed issue, but it was closed by the opener, not the project maintainer: https://github.com/richardpenman/whois/issues/188

The problem is whois is a downstream dependency for my project, so I can't just use a hacky manual fix.

The issue is in the upstream module https://github.com/PythonCharmers/python-future. 

https://github.com/PythonCharmers/python-future/issues/625

The maintainer there doesn't look responsive, but a workaround could be done here.

This totally breaks whois in python 3.12

**Comment by ERHash** [comment]
![image](https://github.com/richardpenman/whois/assets/67637529/b55f3dd0-0e4c-42f6-a61a-14d05abd0dc8)

changing imp (depreciate) by importlib on the __init__ of future can fix it

**Comment by ERHash** [comment]
https://github.com/PythonCharmers/python-future/pull/632

**Comment by richardpenman** [comment]
Fair enough - I'll merge a PR that removes Python 2 support. There was one but it now has merge conflicts and author hasn't responded.

**Comment by richardpenman** [comment]
Have removed python2 support
